### ISSUE TITLE
Invalid warnings on typespecs with bitstrings

### ISSUE THREAD

**Issue by lukaszsamson** [rrrene/credo#1079]
### Precheck

* Proposals for new features should be submitted via: https://github.com/rrrene/credo-proposals
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): 1.7.1-ref.master.4e625ba+uncommittedchanges
* Erlang/Elixir version (`elixir -v`): Erlang/OTP 26 [erts-14.0.2] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit]

Elixir 1.15.6 (compiled with Erlang/OTP 26)
* Operating system: macos

### What were you trying to do?

Function with valid spec (nonempty bitstring and bitstring)
```
@spec some() :: <<_::1, _::_*1>>
def some(), do: <<1::1>>
@spec other() :: <<_::_*1>>
def other(), do: <<>>
```

### Expected outcome

No warning

### Actual outcome

Warning
```
Operation will always return the left side of the expression.
```

**Comment by rrrene** [comment]
Thx for reporting! 👍

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by rrrene** [comment]
This is part of Credo `1.7.2`. :+1:
